File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/04_For_Loop_Exercise/06_Oscars.js
function oscars(input) {
    let pointsNeeded = 1250.50;

    let actorName = input[0];
    let actorPoints = Number(input[1]);
    let evaluatorsCount = Number(input[2]);

    let inputReadCount = evaluatorsCount * 2;

    for (let i = 0; i < inputReadCount; i += 2) {
        let evaluatorName = input[i + 3];
        let evaluatorPoints = Number(input[i + 4]);

        actorPoints += evaluatorName.length * evaluatorPoints / 2;
        // console.log(actorPoints);
        if (actorPoints > pointsNeeded) {
            console.log(`Congratulations, ${actorName} got a nominee for leading role with ${actorPoints.toFixed(1)}!`);
            break;
        }
    }

    if (actorPoints <= pointsNeeded) {
        console.log(`Sorry, ${actorName} you need ${(1250.5 - actorPoints).toFixed(1)} more!`);
    }

}


// oscars(["Zahari Baharov",
//     "205",
//     "4",
//     "Johnny Depp",
//     "45",
//     "Will Smith",
//     "29",
//     "Jet Lee",
//     "10",
//     "Matthew Mcconaughey",
//     "39"]);

File: Python_Courses/Python_Basics/Python_PB_2025_03/04_For_Loop_Exercise/06_Oscars.py
actor_name = input()
starting_points = float(input())
judges_count = int(input())

current_points = starting_points
for _ in range(judges_count):  # [0, 1, 2, 3...]
    judge_name = input()
    judge_points = float(input())

    judge_result = len(judge_name) * judge_points / 2
    current_points += judge_result
    if current_points > 1250.5:
        break

if current_points > 1250.5:
    print(f'Congratulations, {actor_name} got a nominee for leading role with {current_points:.1f}!')
else:
    points_needed = 1250.5 - current_points
    print(f'Sorry, {actor_name} you need {points_needed:.1f} more!')
